### ISSUE TITLE
Focus trap - A11y

### ISSUE THREAD

**Issue by mimarz** [equinor/design-system#1362]
![its a trap.gif](https://images.zenhubusercontent.com/605aea9d482116cb33bf4049/c3ecb761-d9e4-4d7f-81cc-dd6f2a4cd81e)

Introducs missing focus traps on "modal" components for making them correctly accessible.

Relevant components:
* Dialog
* Popover
*....other?

Notes
* https://www.npmjs.com/package/react-focus-lock
* https://www.npmjs.com/package/focus-trap-react

**Comment by akvam** [comment]
When work i starting, US will be added for relevant components.

**Comment by oddvernes** [comment]
Perhaps using the native <Dialog> element can be of use, it has all the focus trapping and other a11y built in, also including ::backdrop and natively existing in a separate layer (so it doesnt need portal)

**Comment by oddvernes** [comment]
Additionally, floating-ui have a [floatingFocusManager](https://floating-ui.com/docs/FloatingFocusManager) we can use with popover

**Comment by akvam** [comment]
No more US on this => closed
